src/main/java/stream/flarebot/flarebot/util/general/FormatUtils.java

```diff
@@ -0,0 +1,178 @@
+     *
+     * @param period Period to format onto the current date
+     * @return The date in a precise format. Example: 2017-10-13 21:56:33.681
+     */
+    public static String formatPrecisely(Period period) {
+        return FormatUtils.preciseFormat.format(DateTime.now(DateTimeZone.UTC).plus(period).toDate());
+    }
+
+    /**
+     * Formats a {@link LocalDateTime} into an easily readable string.
+     * Example: 18th January 2018 15:38:15 UTC.
+     *
+     * @param dateTime The {@link LocalDateTime} to format.
+     * @return A String repressing the time.
+     */
+    public static String formatTime(LocalDateTime dateTime) {
+        LocalDateTime time = LocalDateTime.from(dateTime.atOffset(ZoneOffset.UTC));
+        return time.getDayOfMonth() + GeneralUtils.getDayOfMonthSuffix(time.getDayOfMonth()) + " " + time
+                .format(FormatUtils.timeFormat) + " UTC";
+    }
+
+    /**
+     * Formats duration with the given options.
+     * Example: 16 minutes 40 seconds
+     *
+     * @param duration The duration as a long which will be formatted.
+     * @param durUnit The unit the duration is in.
+     * @param fullUnits Weather not not to return full units. ex days instead of d.
+     * @param append0 Weather or not to append 0s
+     * @return The formatted time.
+     */
+    public static String formatTime(long duration, TimeUnit durUnit, boolean fullUnits, boolean append0) {
+        long totalSeconds = 0;
+        switch (durUnit) {
+            case MILLISECONDS:
+                totalSeconds = duration / 1000;
+                break;
+            case SECONDS:
+                totalSeconds = duration;
+                break;
+            case MINUTES:
+                totalSeconds = duration * 60;
+                break;
+            case HOURS:
+                totalSeconds = (duration * 60) * 60;
+                break;
+            case DAYS:
+                totalSeconds = ((duration * 60) * 60) * 24;
+                break;
+        }
+        long seconds = totalSeconds % 60;
+        long minutes = (totalSeconds / 60) % 60;
+        long hours = (totalSeconds / 3600) % 24;
+        long days = (totalSeconds / 86400);
+        return (days > 0 ? (append0 && days < 10 ? "0" + days : days) + (fullUnits ? " days " : "d ") : "")
+                + (hours > 0 ? (append0 && hours < 10 ? "0" + hours : hours) + (fullUnits ? " hours " : "h ") : "")
+                + (minutes > 0 ? (append0 && minutes < 10 ? "0" + minutes : minutes) + (fullUnits ? " minutes" : "m ") : "")
+                + (seconds > 0 ? (append0 && seconds < 10 ? "0" + seconds : seconds) + (fullUnits ? " seconds" : "s") : "")
+                .trim();
+    }
+
+    /**
+     * Truncates a String to the given length. With ellipses. This truncates from the end.
+     * Note that this will truncate the string to three less then the length, because of the ellipses (The total length of the string would be the provided length).
```

Comment:
This comment is wayyyy too long, line wrap and remove the comma between `length` and `because`